auslib/test/test_db.py

```diff
@@ -1524,6 +1524,14 @@ def testDeleteRelease(self):
         release = self.releases.t.select().where(self.releases.name == 'a').execute().fetchall()
         self.assertEquals(release, [])
 
+    def testDeleteWithRuleMapping(self):
+        self.releases.t.insert().execute(name='d', product='d', data=json.dumps(dict(name="d", schema_version=1, hashFunction="sha512")),
+                                         data_version=1)
+        self.rules.t.insert().execute(rule_id=1, priority=100, version='3.5', buildTarget='d', backgroundRate=100, mapping='d', update_type='z', data_version=1)
+        self.releases.delete({"name": "d"}, changed_by="bill", old_data_version=1)
+        release = self.releases.t.select().where(self.releases.name == 'd').execute().fetchall()
+        self.assertEquals(release, [])
```

Comment:
I think this test is backwards. If Foreign Keys are being enforced, self.releases.delete should raise an Exception rather than letting the Release be deleted, right?